src/App.jsx

```diff
@@ -14,45 +17,107 @@ export const goodsFromServer = [
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+const SORT_REVERSE = 'Reverse';
+
+function getPreparedGoods(goods, { sortField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField.length > 0) {
+    sortField.forEach((elem) => {
+      switch (elem) {
+        case SORT_ALPHABETICALLY:
+
+          return preparedGoods
+            .sort((good1, good2) => good1.localeCompare(good2));
+        case SORT_LENGTH:
+
+          return preparedGoods
+            .sort((good1, good2) => good1.length - good2.length);
+        case SORT_REVERSE:
+
+          return preparedGoods.reverse();
+        default:
+          return preparedGoods;
+      }
+    });
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState([]);
+  const visibleGoods = getPreparedGoods(goodsFromServer, { sortField });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField.indexOf(SORT_ALPHABETICALLY) === -1,
+          })}
+          onClick={() => (
+            sortField.indexOf(SORT_REVERSE) !== -1
+              ? setSortField([SORT_ALPHABETICALLY, SORT_REVERSE])
+              : setSortField([SORT_ALPHABETICALLY])
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField.indexOf(SORT_LENGTH) === -1,
+          })}
+          onClick={() => (
+            sortField.indexOf(SORT_REVERSE) !== -1
+              ? setSortField([SORT_LENGTH, SORT_REVERSE])
+              : setSortField([SORT_LENGTH])
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': sortField.indexOf(SORT_REVERSE) === -1,
+          })}
+          onClick={() => (
+            sortField.indexOf(SORT_REVERSE) === -1
+              ? setSortField([...sortField, SORT_REVERSE])
+              : setSortField(
+                [...sortField].filter(elem => elem !== SORT_REVERSE),
```

Comment:
Looks a bit complicated, let's create a separate function and use it here,  also check comment above